app/client/utils/themes.ts

```diff
@@ -0,0 +1,60 @@
+import type { Theme } from "daisyui";
+
+// You can change these values to other themes provided by daisyUI
+// Or create your own themes and use them here
+export const LIGHT_THEME: Theme = "light";
+export const DARK_THEME: Theme = "dark";
+
+/**
+ * This script handles saving/reading preferred theme from localStorage,
+ * detecting system theme and setting the actual theme as a data attribute on the html element.
+ * It also exposes a function to set the theme which can be called from React components.
+ *
+ * It needs to be injected into the head of the document,
+ * so that it runs before the intial server-side rendered content is displayed,
+ * in order to prevent the flash of incorrect theme.
+ *
+ * It's written as a string, because Helmet doesn't allow any other way to inject scripts.
+ *
+ * This approach is inspired by this blogpost:
+ * https://hangindev.com/blog/avoid-flash-of-default-theme-an-implementation-of-dark-mode-in-react-app
+ */
+export const themeInitScript = `
```

Comment:
I'm not a fan of this approach. Isn't file `app/server/templates/base.html` the one to which we will inject this code? Can't we add this code there?

Reply:
No, `app/server/templates/base.html` is an old template which the current version of the site doesn't use anymore. It will be deleted later. 

Currently the base html template is handled by Reactivated, so it's not easy to access. And even if we could modify it and inline this script there,  it probably wouldn't be possible to use `LIGHT_THEME` and `DARK_THEME` variables anymore (because these variables are also used in `app/tailwind.config.ts`).

I'm not a fan of this approach either, but it was the only one I found which worked properly fully on the client side. An alternative I considered was to store the theme preference on the backend, but it would mean that only signed users can change it, so I didn't go with it.